init.lua

```diff
@@ -177,53 +177,64 @@ function ranks.update_nametag(player)
 	end
 end
 
--- [function] Set player rank
+-- [function] Set player rank TODO
 function ranks.set_rank(player, rank)
+	local name = player
+
 	if type(player) == "string" then
 		player = minetest.get_player_by_name(player)
 	end
```

Comment:
Similarly to lines -55 to -57, some variation of this check should be retained, except modified to ensure that the content of `player` is always a `string` rather than an ObjectRef.

Reply:
OK, will do!